### PR TITLE
Visual Studio 2022 support + drop Visual Studio 2013 support

### PR DESCRIPTION
**Pull request by nullsystem**
- [x] Check Linux works fine with PR #70 
- [x] Redo the commits a bit so CE's works are credited properly
- [x] Update README/docs

* Can now compile, link, and run using VS2022
* VS2013 no longer needs to be installed + removed support
    * Removed support for Windows XP SP1???
* From SDK2013-CE/VS2022 works:
    * particle.lib - https://developer.valvesoftware.com/wiki/Compiling_under_VS2022 
        * https://github.com/SCell555/Source-PlusPlus/commit/b44ac3d0b3333ad2d2af85fa5422fca52ef18b10
    * memoverride.cpp - SDK2013-CE
    * vpc.exe - https://github.com/Nbc66/source-sdk-2013-ce/commit/3b4bf98f15169d79ccfd7ec8fd6c1fac3131f5ad [Cherry-picked]
* Fixed compile errors, suppressed warnings related to upgrading from VS2013 to VS2022
* Edited the strings on vpc.exe to say Visual Studio 2022 instead of 2013
* Basically the first part of integrating source-sdk-2013-ce changes into neo, more can start refitting neo + ce changes after this one
* Extra .gitignores to prevent committing in bsp, pdb
* Most of the line changed are on `memoverride.cpp` which is just copied over from SDK2013-CE, most of it is really just around 80~ added/deleted lines
* Bump minimum instruction set from SSE to SSE2

### CODE REVIEW

**Comment by Agiel** [comment]
Can confirm that this works. I was able to set up everything from scratch with Visual Studio Community 2022.
